src/odf/metadata/src/formats/did_pkh.rs

```diff
@@ -0,0 +1,163 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use multiformats::stack_string::AsStackString;
+use ssi_caips::caip10;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+pub const DID_PKH_PREFIX: &str = "did:pkh:";
+
+// Based on ssi-caips/src/caip10.rs
+const CAIP_10_CHAIN_ID_MAX_LENGTH: usize = 41;
+pub const CAIP_10_ACCOUNT_ADDRESS_MAX_LENGTH: usize = 64;
+
+pub const MAX_DID_PKH_STRING_REPR_LEN: usize = DID_PKH_PREFIX.len()
+    + CAIP_10_CHAIN_ID_MAX_LENGTH
+    + ":".len()
+    + CAIP_10_ACCOUNT_ADDRESS_MAX_LENGTH;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/// Represents DID in `did:pkh` method.
+///
+/// Format: `did:pkh:<account_id(CAIP-10)>`
+///
+/// Example: `did:pkh:eip155:1:0xab16a96D359eC26a11e2C2b3d8f8B8942d5Bfcdb`
+#[derive(Clone, PartialEq, Eq, Hash)]
+pub struct DidPkh {
+    caip10_account_id: caip10::BlockchainAccountId,
+}
```

Comment:
I propose isolating entire PKH stuff in ODF level under a feature flag.

Reply:
Feature gate (did-pkh) added

Done